transports/onion/Cargo.toml

```diff
@@ -0,0 +1,33 @@
+[package]
+name = "libp2p-onion"
+version = "0.1.0"
+edition = "2021"
+license = "MIT"
+resolver = "2"
+
+
+[dependencies]
+libp2p-core = { version = "0.36.0", path = "../../core", default-features = false }
+futures = "0.3"
+arti-client = { version = "0.6", default-features = false }
+thiserror = "1"
+tor-rtcompat = "0.6"
+tokio-crate = { package = "tokio", version = "1", optional = true, default-features = false }
+async-std-crate = { package = "async-std", version = "1", optional = true, default-features = false }
+
+[dev-dependencies]
+libp2p = { version = "0.49", path = "../../" }
+tokio-crate = { package = "tokio", version = "1", features = ["rt", "macros"] }
+async-std-crate = { package = "async-std", version = "1", features = ["attributes"] }
+
+
+[features]
+default = ["async-std", "native-tls"]
+tokio = ["arti-client/tokio", "dep:tokio-crate"]
+async-std = ["arti-client/async-std"]
+native-tls = ["arti-client/native-tls"]
+rustls = ["arti-client/rustls"]
+
+[package.metadata.docs.rs]
+rustdoc-args = ["--cfg", "docsrs"]
+rustc-args = ["--cfg", "docsrs"]
```

Comment:
What is this needed for?

Reply:
I can't really try this, however I used this in my other crates in order to enable the display of features in the docs.

Reply:
What this does is tell docs.rs to pass the `docsrs` conditional compilation flag to the rust compiler, meaning any code that is guarded with `cfg(docsrs)` will now be included in the build.

We use this in the `libp2p` root crate to conditionally include the `doc` attribute: https://github.com/libp2p/rust-libp2p/blob/04b07c4d14a0a6ec350fc21261a33d859fc71289/src/lib.rs#L53

We need to do this conditionally because the `#[doc(cfg)]` attribute is [not yet stable](https://github.com/rust-lang/rust/issues/43781). docs.rs runs all their build pipelines on nightly so we can use it there.

Now, interestingly, we don't set this in our `libp2p` crate so all of these `cfg_attr` attributes have been doing nothing all this time :)

In this crate, it is used here: https://github.com/umgefahren/rust-libp2p/blob/6f4bca5bbfd141bc18d9d931c5d28b50a212f0fe/transports/onion/src/provider.rs#L31-L36

And will tell docs.rs to include a little "only available with feature `tokio`" tag next to the item.

Reply:
This is actually something that should be fixed. I'll open an issue about it.